doc/index.rst

```diff
@@ -54,6 +54,51 @@ Your contributions are welcome!  Simply fork the `GitHub repository <https://git
 
 .. _pull request: https://github.com/python-control/python-control/pulls
 
+The following details suggested steps for making your own contributions to the project using GitHub
+
+1. Fork on GitHub: login/create an account and click Fork button at the top right corner of https://github.com/python-control/python-control/.
+
+2. Clone to computer (Replace [you] with your Github username)::
+
+    git clone https://github.com/[you]/python-control.git
+    cd python_control
+
+3. Set up remote upstream::
+
+    git remote add upstream https://github.com/python-control/python-control.git
+
+4. Start working on a new issue or feature by first creating a new branch with a descriptive name::
+
+    git checkout -b <my-new-branch-name>
+
+5. Write great code. Suggestion: write the tests you would like your code to satisfy before writing the code itself. This is known as test-driven development.
+
+6. Run tests and fix as necessary until everything passes::
+
+    pytest -v
+
+  (for documentation, run ``make html`` in ``doc`` directory)
+
+7. Commit changes::
+
+    git add <changed files>
+    git commit -m "commit message"
+
+8. Update & sync your local code to the upstream version on Github before submitting (especially if it has been awhile)::
+
+    git checkout master; git fetch --all; git merge upstream/master; git push
```

Comment:
you probably don't want `merge` but `rebase` here to avoid accidental push/pulls.

Reply:
I'd rather put the `git push` into a separate line and check the result after the merge/rebase before proceeding.

Rule of thumb: "Never rebase master". Not even your own origin. Of course the rebase vs merge argument is irrelevant if you always develop in feature branches and don't commit anything into master that is not on upstream. Then the merge will always be a fast-forward.

Reply:
i put the push on a separate line.  maybe better then to leave as-is in case noobs commit to master?